posthog/models/feedback/survey.py

```diff
@@ -62,3 +52,16 @@ class Meta:
     end_date: models.DateTimeField = models.DateTimeField(null=True)
     updated_at: models.DateTimeField = models.DateTimeField(auto_now=True)
     archived: models.BooleanField = models.BooleanField(default=False)
+
+
+@mutable_receiver(post_save, sender=Survey)
+def update_surveys_opt_in(sender, instance, **kwargs):
+    active_surveys_count = Survey.objects.filter(
+        team_id=instance.team_id, start_date__isnull=False, end_date__isnull=True, archived=False
+    ).count()
+    if active_surveys_count > 0 and instance.team.surveys_opt_in is (False or None):
+        instance.team.surveys_opt_in = True
+        instance.team.save()
```

Comment:
safer when we explicitly mention to only update our relevant field.

```suggestion
        instance.team.save(update_fields=["surveys_opt_in"])
```